src/server/routes/admin/nonces.ts

```diff
@@ -0,0 +1,169 @@
+      lastUsedNonce: "8",
+      recycledNonces: ["6", "7"],
+      chainId: 80001,
+    },
+  ],
+};
+
+export async function getNonceDetailsRoute(fastify: FastifyInstance) {
+  fastify.route<{
+    Querystring: Static<typeof walletWithAddressParamSchema>;
+    Reply: Static<typeof responseBodySchema>;
+  }>({
+    method: "GET",
+    url: "/admin/nonces",
+    schema: {
+      summary: "Get transaction details",
+      description: "Get raw logs and details for a transaction by queueId.",
+      tags: ["Admin"],
+      operationId: "transactionDetails",
+      querystring: walletWithAddressParamSchema,
+      response: {
+        ...standardResponseSchema,
+        [StatusCodes.OK]: responseBodySchema,
+      },
+      hide: true,
+    },
+    handler: async (request, reply) => {
+      const { walletAddress, chain } = request.query;
+      const result = await getNonceDetails({ walletAddress, chainId: chain });
+
+      reply.status(StatusCodes.OK).send({
+        result,
+      });
+    },
+  });
+}
+
+export const getNonceDetails = async ({
+  walletAddress,
+  chainId,
+}: {
+  walletAddress?: string;
+  chainId?: string;
+} = {}) => {
+  const lastUsedNonceKeys = await getLastUsedNonceKeys(walletAddress, chainId);
+
+  const result = await Promise.all(
+    lastUsedNonceKeys.map(async (key) => {
+      const { chainId, walletAddress } = splitLastUsedNonceKey(key);
+
+      const [lastUsedNonce, sentNonces, recycledNonces, onchainNonce] =
+        await Promise.all([
+          redis.get(lastUsedNonceKey(chainId, walletAddress)),
+          redis.smembers(sentNoncesKey(chainId, walletAddress)),
+          redis.smembers(recycledNoncesKey(chainId, walletAddress)),
+          getOnchainNonce(chainId, walletAddress),
+        ]);
+
+      return {
+        walletAddress,
+        onchainNonce: onchainNonce.toString(),
+        lastUsedNonce: lastUsedNonce ?? "0",
+        sentNonces,
+        recycledNonces,
+        chainId,
+      };
+    }),
+  );
+
+  return result;
+};
+
+/**
+ * Get all lastUsedNonce keys from Redis, based on the provided wallet address and chain.
+ * @param walletAddress Wallet address to filter by (optional)
+ * @param chainId chainId to filter by (optional)
+ *
+ * @returns Array of lastUsedNonce keys for wallet address and chain combinations
+ */
+const getLastUsedNonceKeys = async (
+  walletAddress?: string,
+  chainId?: string,
+): Promise<string[]> => {
+  const keys = await redis.keys(
+    `nonce:${chainId ?? "*"}:${
+      walletAddress ? normalizeAddress(walletAddress) : "*"
+    }`,
+  );
+
+  return keys;
+};
+
+const getOnchainNonce = async (chainId: number, walletAddress: string) => {
+  const rpcRequest = getRpcClient({
+    client: thirdwebClient,
+    chain: await getChain(chainId),
+  });
+
+  // The next unused nonce = transactionCount.
+  const transactionCount = await eth_getTransactionCount(rpcRequest, {
+    address: walletAddress,
+  });
+
+  const onchainNonce = transactionCount + 1;
```

Comment:
You want `transactionCount - 1` here I believe.

The naming of `onchain nonce` is a bit ambiguous. Do you want:
1. The _next unused_ nonce?
2. The _last used_ nonce?

I think you probably want option 2 since it matches the db nonce we're querying (the last used db nonce).